src/games/strategy/triplea/ui/RouteOptimizer.java

```diff
@@ -0,0 +1,250 @@
+package games.strategy.triplea.ui;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.awt.Point;
+import java.awt.geom.Line2D;
+import java.awt.geom.Point2D;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import games.strategy.triplea.ui.mapdata.MapData;
+
+public class RouteOptimizer {
```

Comment:
Looks like a good decomposition :+1:   Noting that this class is much easier to test than MapRouteDrawer, and also makes MapRouteDrawer since we can mock the logic here and return hardcoded values so we do not have to worry about the calculations when testing MapRouteDrawer.